0000-interledger-tokens/0000-interledger-tokens.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Interledger Tokens
+draft: 1
+---
+# Interledger Tokens
+
+An **Interledger Token** is a special form of JSON Web Token that can be used to initiate an Interledger payment. The token provides proof that the payment has been pre-authorized by the payer. By using these tokens new use cases for "debit pull" style payments such as recurring subscriptions or offline authorization are possible.
+
+## Background
+
+TODO
+
+## Motivation
+
+TODO
+
+## Terminology
+
+The following terms are used throughout this specification. Where they are used their meaning as follows:
+
+### Payer
+
+The entity whose funds are being spent when the token is used to initiate a payment.
+
+### Payer's Interledger Service Provider (ILSP)
+
+The entity that accepts and processes the token and payment request. This may be the payer or they may delegate it to some other entity that is able to make payments on their behalf.
+
+### Payee
+
+The entity that is requesting payment using the token.
+
+## The Interledger Token
+
+An **Interledger Token** is a special form of a JSON Web Token (JWT) [RFC7519](https://tools.ietf.org/html/rfc7519) that can be submitted to a participant on the Interledger network to request that they initiate a new payment via ILP. The content of the token describes how much to pay, who to pay and provides information for the entity processing the token to verify that the payment was authorized by the owner of the funds being sent.
```

Comment:
Have you looked at macaroons at all? I think they'd be pretty cool for something like this because whoever you give it to could add further "caveats" and pass it on more

Reply:
Not in depth admittedly. JWT is a now a pretty well bedded down standard. It has had a lot of criticism but I believe this spec addresses all of them by using a very limited profile of JWT.

Reply:
I have to say that I never really regained respect for JWTs after the [`none` bug](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/)...

Reply:
I think that's a little unfair. That was a foot-gun sure, but it also only materialized in bad implementations that ignored the advice **in the spec** to use different keys for different algorithms. 

For me, the whole value of using a standard like JWT is it has lots of eyes on it so the vuln was found and fixed. It is anti-fragile.